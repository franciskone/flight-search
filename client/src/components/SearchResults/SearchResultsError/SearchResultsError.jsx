import React from 'react';
import PropTypes from 'prop-types';
import BpkCard from 'bpk-component-card';
import BpkText from 'bpk-component-text';
import BpkButton from 'bpk-component-button';


import { styleGetter } from '../../../util';
import STYLE from './SearchResultsError.scss';

const c = styleGetter(STYLE);

const SearchResultsError = ({ message, action, buttonLabel }) => (
  <BpkCard className={c('SearchResultsError')}>
    <BpkText
      className={c('SearchResultsErrorMessage')}
      textStyle="xl"
    >
      {message}
    </BpkText>
    
    <BpkButton
      large
      onClick={action}
    >
      {buttonLabel}
    </BpkButton>
  </BpkCard>
);

SearchResultsError.propTypes = {
  message: PropTypes.string.isRequired,
  buttonLabel: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default SearchResultsError;
