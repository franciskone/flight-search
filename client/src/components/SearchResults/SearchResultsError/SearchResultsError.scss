@import '~bpk-mixins';

.SearchResultsError {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 15rem;
  margin: $bpk-spacing-md $bpk-spacing-sm;

  &Message {
    color: $bpk-color-red-500;
    margin-bottom: $bpk-spacing-md;
    text-align: center;
  }
}
