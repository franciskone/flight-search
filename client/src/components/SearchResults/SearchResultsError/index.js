import SearchResultsError from './SearchResultsError';

export default SearchResultsError;
