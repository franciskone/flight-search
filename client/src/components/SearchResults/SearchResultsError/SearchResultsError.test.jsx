import React from 'react';
import TestRenderer from 'react-test-renderer';

import SearchResultsError from './SearchResultsError';

describe('SearchResultsError', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(
      <SearchResultsError
        message="test message"
        buttonLabel="test label"
        action={() => null}
      />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
