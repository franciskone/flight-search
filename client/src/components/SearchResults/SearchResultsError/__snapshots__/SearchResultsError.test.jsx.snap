// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsError should render correctly 1`] = `
<div
  className="bpk-card bpk-card--padded SearchResultsError"
  role="button"
>
  <span
    className="bpk-text bpk-text--xl SearchResultsErrorMessage"
  />
  <button
    className="bpk-button bpk-button--large"
    disabled={false}
    onClick={null}
    type="button"
  />
</div>
`;
